ext/RMagick/extconf.rb

```diff
@@ -123,31 +123,73 @@ def exit_failure(msg)
       exit(1)
     end
 
+    def detect_imagemagick_packages(packages)
+      existed_packages = packages.map do |package|
+        if system "pkg-config --exists #{package}"
+          package
+        end
+      end
+      existed_packages.compact
+    end
+
+    def detect_im6_packages
+      packages = %w[
+        ImageMagick-6.Q64HDRI
+        ImageMagick-6.Q32HDRI
+        ImageMagick-6.Q16HDRI
+        ImageMagick-6.Q8HDRI
+        ImageMagick-6.Q64
+        ImageMagick-6.Q32
+        ImageMagick-6.Q16
+        ImageMagick-6.Q8
+      ]
+      @detect_im6_packages ||= detect_imagemagick_packages(packages)
+    end
+
+    def detect_im7_packages
+      packages = %w[
+        ImageMagick-7.Q64HDRI
+        ImageMagick-7.Q32HDRI
+        ImageMagick-7.Q16HDRI
+        ImageMagick-7.Q8HDRI
+        ImageMagick-7.Q64
+        ImageMagick-7.Q32
+        ImageMagick-7.Q16
+        ImageMagick-7.Q8
+      ]
+      @detect_im7_packages ||= detect_imagemagick_packages(packages)
+    end
+
+    def im6_package_exists?
+      package = detect_im6_packages
+      package.empty? ? false : true
+    end
+
+    def im7_package_exists?
+      package = detect_im7_packages
+      package.empty? ? false : true
+    end
+
     def determine_imagemagick_package
-      packages = `pkg-config --list-all`.scan(/(ImageMagick\-[\.A-Z0-9]+) .*/).flatten
+      packages = [detect_im7_packages, detect_im6_packages].flatten
 
-      # For ancient version of ImageMagick 6 we need a different regex
       if packages.empty?
-        packages = `pkg-config --list-all`.scan(/(ImageMagick) .*/).flatten
+        # ImageMagick 6.7 does not have package file like ImageMagick-6.Q16.pc
```

Comment:
Do we want to remove 6.7 support for RMagick 5.0?

Reply:
I think we can remove it after end of life of 14.04 LTS.

https://github.com/rmagick/rmagick/issues/474